src/webgpu/api/validation/attachment_compatibility.spec.ts

```diff
@@ -18,6 +16,60 @@ import {
 import { ValidationTest } from './validation_test.js';
 
 const kColorAttachmentCounts = range(kMaxColorAttachments, i => i + 1);
+const kColorAttachments = kColorAttachmentCounts
+  .map(count => {
+    // generate cases with 0..1 null attachments at different location
+    // e.g. count == 2
+    // [
+    //    [1, 1],
+    //    [0, 1],
+    //    [1, 0],
+    // ]
+    // 0 means null attachment, 1 means non-null attachment, at the slot
+
+    // Special cases: we need at least a color attachment, when we don't have depth stencil attachment
+    if (count === 1) {
+      return [[1]];
+    }
+    if (count === 2) {
+      return [
+        [1, 1],
+        [0, 1],
+        [1, 0],
+      ];
+    }
+
+    // [1, 1, ..., 1]: all color attachment are used
+    let result = [new Array<number>(count).fill(1)];
+
+    // [1, 0, 1, ..., 1]: generate cases with one null attachment at different locations
+    result = result.concat(
+      range(count, i => {
+        const r = new Array<number>(count).fill(1);
+        r[i] = 0;
+        return r;
+      })
+    );
+
+    // [1, 0, 1, ..., 0, 1]: generate cases with two null attachments at different locations
+    result = result.concat(
+      range(count - 1, i => {
+        const cases = [] as number[][];
+        for (let j = i + 1; j < count; j++) {
+          const r = new Array<number>(count).fill(1);
+          r[i] = 0;
+          r[j] = 0;
+          cases.push(r);
+        }
+
+        return cases;
+      }).flat()
+    );
+
+    return result;
+  })
+  .flat() as number[][];
```

Comment:
maybe make this boolean[][] ?

then later, `map` can just do `i ? 'rgba8unorm' : null`

or, since we always use rgba8unorm, we could just output that now, instead of translating later

Reply:
use boolean (still needs to translate to color formats, color targets, or color attachments)